### ISSUE TITLE
Steps 8 and 10 are trying to generate dbxref for ontology term

### ISSUE THREAD

**Issue by dsenalik** [tripal/tripal_analysis_expression#417]
Errors occuring loading a biosample xml file. These arise when the code attempts to generate a dbxref for an ontology term.
Based on this comment this should not be happening, but I don't see the code that determines that. I can get past these by checking for a null in two places. The question is whether that is correct, or I should check earlier. Investigation to follow...

```
     // Only do an insert if this dbxref doesn't already exist in the databse.
     // and this dbxref is from a Dbxref attribute not an Ontology_term attr.
```
The test biosample xml is from https://www.ncbi.nlm.nih.gov/biosample/28751899

error at step 8
```
SQLSTATE[23502]: Not null violation: 7 ERROR:  null value in column "db_id" of relation "dbxref" violates not-null constraint
DETAIL:  Failing row contains (444922, null, Healthy panax notoginseng, , null).
[site http://default] [TRIPAL ERROR] [TRIPAL_JOB] SQLSTATE[23502]: Not null violation: 7 ERROR:  null value in column "db_id" of relation "dbxref" violates not-null constraintDETAIL:  Failing row contains (444922, null, Healthy panax notoginseng, , null).
```

error at step 10
```
SQLSTATE[23502]: Not null violation: 7 ERROR:  null value in column "dbxref_id" of relation "biomaterial_dbxref" violates not-null constraint
DETAIL:  Failing row contains (2329, 1307, null).
[site http://default] [TRIPAL ERROR] [TRIPAL_JOB] SQLSTATE[23502]: Not null violation: 7 ERROR:  null value in column "dbxref_id" of relation "biomaterial_dbxref" violates not-null constraintDETAIL:  Failing row contains (2329, 1307, null).
```

**Comment by dsenalik** [comment]
The cause is that the xml for this biosample looks like this
```
    <Ids>
      <Id db="BioSample" is_primary="1">SAMN28751899</Id>
      <Id db_label="Sample name">Healthy panax notoginseng</Id>
      <Id db="SRA">SRS13217750</Id>
    </Ids>
```
and the code tries to process the middle line as a dbxref.

lol, I think I can fix it right around this line in the code
```// TODO: what to do.```

**Comment by spficklin** [comment]
Fixed with PR #418
